chain/ethereum/src/network_indexer/ethereum.graphql

```diff
@@ -0,0 +1,86 @@
+""" Block is an Ethereum block."""
+type Block @entity {
+  id: ID!
+
+  """The number of this block, starting at 0 for the genesis block."""
+  number: BigInt!
+
+  """The block hash of this block."""
+  hash: Bytes!
+
+  """The parent block of this block."""
+  parent: Block
+
+  """The block nonce, an 8 byte sequence determined by the miner."""
+  nonce: Bytes!
+
+  """The keccak256 hash of the root of the trie of transactions in this block."""
+  transactionsRoot: Bytes!
+
+  """The number of transactions in this block."""
+  transactionCount: Int!
+
+  """The keccak256 hash of the state trie after this block was processed."""
+  stateRoot: Bytes!
+
+  """The keccak256 hash of the trie of transaction receipts in this block."""
+  receiptsRoot: Bytes!
+
+  # """The account that mined this block."""
+  # miner: Account!
+
+  """An arbitrary data field supplied by the miner."""
+  extraData: Bytes!
+
+  """The maximum amount of gas that was available to transactions in this block."""
+  gasLimit: BigInt!
+
+  """The amount of gas that was used executing transactions in this block."""
+  gasUsed: BigInt!
+
+  """The unix timestamp at which this block was mined."""
+  timestamp: BigInt!
+
+  """
+  A bloom filter that can be used to check if a block may
+  contain log entries matching a filter.
+  """
+  logsBloom: Bytes!
+
+  """The hash that was used as an input to the PoW process."""
+  mixHash: Bytes!
+
+  """A measure of the difficulty of mining this block."""
+  difficulty: BigInt!
+
+  """The sum of all difficulty values up to and including this block."""
+  totalDifficulty: BigInt!
+
+  """Whether the block is an ommer."""
+  isOmmer: Boolean!
+
+  """The number of ommers (AKA uncles) associated with this block."""
+  ommerCount: Int!
+
+  # """
+  # A list of ommer (AKA uncle) blocks associated with this block.
+  # """
+  ommers: [Block]!
+
+  """The keccak256 hash of all the ommers (AKA uncles) associated with this block."""
+  ommerHash: Bytes!
+
+  # """
+  # A list of transactions associated with this block.
+  # """
+  # transactions: [Transaction!]
+
+  # """The logs emitted in this block."""
+  # logs: [Log!]!
+
+  """Size of the block in bytes."""
+  size: BigInt
```

Comment:
I did this because it is optional here: https://tomusdrw.github.io/rust-web3/web3/types/struct.Block.html#structfield.size. I've seen on Etherscan that uncles don't report their size, so maybe that's why it is like that in rust-web3.

Reply:
Makes sense.